robotoff/prediction/ocr/packaging.py

```diff
@@ -0,0 +1,217 @@
+from typing import Optional, Union
+
+import cachetools
+from lark import Discard, Lark, Transformer
+
+from robotoff import settings
+from robotoff.prediction.ocr.dataclass import OCRResult, get_text
+from robotoff.prediction.ocr.grammar import (
+    generate_terminal_symbols_file,
+    normalize_string,
+)
+from robotoff.prediction.types import Prediction, PredictionType
+from robotoff.taxonomy import TaxonomyType
+from robotoff.types import PackagingElementProperty
+from robotoff.utils import load_json
+from robotoff.utils.text import strip_consecutive_spaces
+
+# Set of shapes to exclude when the material or recycling instructions were
+# not detected along the shape
+# Avoid many false positive
+SHAPE_ONLY_EXCLUDE_SET = {
+    "en:tray",  # coque
+    "en:packaging",  # emballage
+    "en:backing",  # plaque
+    "en:packet",  # pack
+    "en:string",  # ficelle
+    "en:bowl",  # bol
+    "en:sheet",  # feuilles
+    "en:net",  # filet
+    "en:pouch-flask",  # pouch, english word
+    "en:spoon",  # most of the spoon are false positive
+    "en:tablespoon",
+    "en:mold",  # moule
+    None,
+}
+
+
+def generate_packaging_lark_file(lang: str):
+    generate_terminal_symbols_file(
+        output_path=settings.GRAMMARS_DIR / f"terminal_packaging_shape_{lang}.lark",
+        lang=lang,
+        terminal_name="PACKAGING_SHAPES",
+        taxonomy_type=TaxonomyType.packaging_shape,
+        terminal_priority=1,
+        ignore_ids={"en:unknown"},
+    )
+    generate_terminal_symbols_file(
+        output_path=settings.GRAMMARS_DIR / f"terminal_packaging_material_{lang}.lark",
+        lang=lang,
+        terminal_name="PACKAGING_MATERIALS",
+        taxonomy_type=TaxonomyType.packaging_material,
+        terminal_priority=1,
+        ignore_ids={"en:unknown"},
+    )
```

Comment:
why don't we have recycling also ?

Reply:
Because recycling instructions were generated manually, to split between "VERB" and "NON_VERB".
It also required a few manual fixes. I've added a comment in the file to explain this.